python/lsst/alert/stream/alertConsumer.py

```diff
@@ -84,6 +84,11 @@ def poll(self, decode=False, verbose=True):
                     else:
                         ast_msg = literal_eval(str(msg.value(), encoding='utf-8'))
                         return ast_msg
+        else:
+            try:
+                raise EopError(msg)
+            except AttributeError:
+                pass
```

Comment:
Hmm, I'm confused by this logic.

At line 77, you have `if msg is not None`. That means that if you enter the `else` at line 87, `msg` *must be* `None`.

If `msg` is `None`, you know that feeding it to `EopError` is going to thrown (`None.topic` is an immediate `AttributeError`), so you needn't even bother trying.

In short, as I read it, the above is equivalent to:

```
if msg is not None:
    # do some stuff
else:
   pass
```

If I'm right, you might (at a minimum) leave out the `else`. But then I'm confused... are you expecting `msg` to be `None` sometimes? Is that a case we have to handle with something other than a `pass`?

Reply:
Yes, I agree, it makes no sense, I don't know why it worked.  It was working to print status for the case of an empty stream, which still makes no sense.  I reverted it back, and it works the old way now.  I think I just changed it while debugging the Kafka version upgrade and left it in.

Reply:
Ok, reverting it back did not work.  Using `if msg is not None:` instead of` if msg:` works as it did before.  Without that it doesn't give me an empty stream status, as it did with the previous Kafka version.  I thought they were the same.  I guess it should be explicit.